neofs-testlib/neofs_testlib/env/env.py

```diff
@@ -178,6 +199,44 @@ def deploy_rest_gw(self):
         self.rest_gw.start()
         allure.attach(str(self.rest_gw), "rest_gw", allure.attachment_type.TEXT, ".txt")
 
+    @allure.step("Deploy neofs contract")
+    def deploy_neofs_contract(self):
+        if len(self.inner_ring_nodes) < 1:
+            raise RuntimeError(
+                "There should be at least a single IR instance configured(not started) to deploy neofs contract"
+            )
+        neo_go = self.neo_go()
+        neo_go.nep17.transfer(
+            "GAS",
+            self.default_wallet.address,
+            f"http://{self.main_chain.rpc_address}",
+            from_address=self.inner_ring_nodes[-1].alphabet_wallet.address,
+            amount=9000,
+            force=True,
+            wallet_config=self.main_chain.neo_go_config,
+        )
+        ir_alphabet_pubkey_from_neogo = wallet_utils.get_last_public_key_from_wallet_with_neogo(
+            self.neo_go(), self.default_wallet.path
```

Comment:
with alphabet wallet here I got:
```
[MainThread] 21:01:12 [INFO] Command: ./neo-go  contract deploy --manifest './neofs-contract/neofs/config.json' --rpc-endpoint 'http://localhost:56972' --wallet-config 'env_files/neofs-env-2024-07-23-21-00-56/main_chain_neo_go_config_rjnxgzhoff.yml' --force --timeout '10s' [ true ffffffffffffffffffffffffffffffffffffffff [ 02f8feae7df6bebc80c69301bb25e0379eb26c3257d7adc26bc5f9148c093d1990 ] [ InnerRingCandidateFee 10 WithdrawFee 10 ] ] --in './neofs-contract/neofs/neofs_contract.nef'
Error:
return code: 1 
Output: Insufficient funds (-511) - insufficient funds
```
Although it has funds, since the transfer command from alphabet address works. In dev-env repo we also deploy neofs contract with a separate wallet. Not sure why.

Reply:
@evgeniiz321, there can be some errors or unexpected zero balances or magic things, but the third arg _must_ be an alphabet key(s): https://github.com/nspcc-dev/neofs-contract/blob/99fb86c35a48ed12017423aa4fee13f7d07fa4c0/contracts/neofs/contract.go#L81C28-L81C39, otherwise you cannot pass this ("this method must be invoked by alphabet") error

dev-env also uses alphabet key: https://github.com/nspcc-dev/neofs-dev-env/blob/b355965e7f0a6c3b99b3da152bf5c09c595ab955/Makefile#L122 (you can dump key from services/ir/az.json and check that it is the same `02b3622bf4017bdfe317c58aed5f4c753f206b7db896046fa7d774bbc4bf7f8dc2`)

you can use a different _wallet_ that signs deploying TX (currently it may differ from the third arg) if it is convenient or you can also use the alphabet _wallet_. in any case, 1. this wallet _must_ have some GAS, cause it works this way, 2. alphabet public key _must_ be specified if you want balances to work